accepted/2021/Globalization/GlobalizationSize.md

```diff
@@ -0,0 +1,81 @@
+
+### ICU4X
+
+It is interesting enough, there is a newly launched open-source project called [ICU4X](https://github.com/unicode-org/icu4x#icu4x----) for introducing globalization support libraries (similar to ICU). ICU4X has the exact goals we need to achieve with our concerned scenarios.
+
+```
+ICU4X will provide an ECMA-402-compatible API surface in the target client-side platforms, including the web platform, iOS, Android, WearOS, WatchOS, Flutter, and Fuchsia, supported in programming languages including Rust, JavaScript, Objective-C, Java, Dart, and C++.
+
+The design goals of ICU4X are:
+- Small and modular code
+- Pluggable locale data
+- Availability and ease of use in multiple programming languages
+- Written by i18n experts to encourage best practices
+```
+
+That is exactly what we need (or what we are trying to achieve).
+Eric Erhardt already did some investigation and arranged a meeting with the project committee to learn more about the project and the status of the project. It was very helpful getting in touch and learning more about this project. It is a promising project managed by experts and using the CLDR data which means it still sticking with the standards.
+
+The catch here is ICU4X still in the early stages and still not providing all functionality needed by .NET for globalization support. The project committee is willing to get and prioritize requests from different parties. We already communicated in the meeting what functionality .NET currently using from ICU4C and want to see that supported in ICU4X. For example, collation support is one of the topics we brought to the committee's attention.
+
+### Plan Suggestion
+
+ICU4X is the promising path we should pursue to address all concerned scenarios. We can wait a little bit more to get the needed missing functionality implemented in the project and integrate it to .NET as we did with ICU4C.
+
+- We need to look more at what is currently supported functionality and what is missing so we can have the full list of features we need to get.
+- We need to be in touch with the ICU4X committee communicating our requests and understanding when such features can be available.
+- Need to look if we can associate some resources to help with that project especially with the missing features we need in the .NET.
+- Need to look at the scope of work to integrate ICU4X to .NET runtime.
+- Need to look if we can have a process creating different NuGet packages from the ICU4X repo with data and code customization.
+
+.NET 7.0 would be the best to start invest in that and try to start consuming at least the available parts of ICU4X.
+
+### Alternative Plans
```

Comment:
Are there other alternatives? What about mixing native APIs with custom code/data or using custom compression for the data as another options?

Reply:
what you mean by custom comparison? Do you mean we implement Unicode Collation Algorithm ourselves? I would strongly oppose that as it is complicated and will need a huge effort to maintain it. ICU spent long time refining it to get it to the current shape. or you have other ideas?

Reply:
"custom compression", not "custom comparison".

I was actually wondering whether the data are compressed and how. Maybe that would be an explanation for the difference in the data files size between the Andorid/iOS builds if a different compression level/library was used.

Mixing native APIs (as mentioned by @spouliot above) may actually be a way to explore. Sure, you may get slightly different behavior but likely the difference will be smaller than NLS->ICU switch since the underlying iOS API use the ICU data.

Reply:
oh, thank you. I misread it. sorry about that.